packages/base-driver/lib/basedriver/helpers.js

```diff
@@ -18,13 +18,14 @@ const CACHED_APPS_MAX_AGE = 1000 * 60 * 60 * 24; // ms
 const APPLICATIONS_CACHE = new LRU({
   maxAge: CACHED_APPS_MAX_AGE, // expire after 24 hours
   updateAgeOnGet: true,
-  dispose: async (app, {fullPath}) => {
-    if (!await fs.exists(fullPath)) {
-      return;
-    }
-
-    logger.info(`The application '${app}' cached at '${fullPath}' has expired`);
-    await fs.rimraf(fullPath);
+  dispose: (app, {fullPath}) => {
+    logger.info(`The application '${app}' cached at '${fullPath}' has ` +
+      `expired after ${CACHED_APPS_MAX_AGE}ms`);
+    setTimeout(async () => {
+      if (fullPath && await fs.exists(fullPath)) {
+        await fs.rimraf(fullPath);
+      }
```

Comment:
```suggestion
      if (fullPath) {
        await fs.rimraf(fullPath);
      }
```

checking for file existence is not necessary, since `rimraf` shouldn't care either way.

Reply:
ok